File: backend/app/schemas/analytics.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class PerformanceMetrics(BaseModel):
    sharpe_ratio: float
    max_drawdown: float
    win_rate: float
    total_trades: int
    total_pnl: float
    start_date: Optional[datetime]
    end_date: Optional[datetime]

File: frontend/src/services/analytics.ts
import apiClient from './client';

export interface PerformanceMetrics {
    sharpe_ratio: number;
    max_drawdown: number;
    win_rate: number;
    total_trades: number;
    total_pnl: number;
    start_date?: string;
    end_date?: string;
}

export const getPerformanceMetrics = async (startDate?: string, endDate?: string): Promise<PerformanceMetrics> => {
    const params: any = {};
    if (startDate) params.start_date = startDate;
    if (endDate) params.end_date = endDate;

    const response = await apiClient.get('/analytics/performance', { params });
    return response.data;
};
